### ISSUE TITLE
User control panel > Add user issues with validation errors

### ISSUE THREAD

**Issue by stevepiercy** [plone/volto#6197]
**Describe the bug**

When adding a user in the User control panel, and when any error occurs, the submit button doubles up.

Additionally one must scroll the panel to view any error messages, and it is not obvious that there may be a problem.

**To Reproduce**

1. Login as admin/admin on https://demo.plone.org/
2. Go to the User control panel https://demo.plone.org/controlpanel/users
3. Click + to add a new user.
4. Fill in details but leave the password field blank.
5. Submit the form.
6. Scroll to the top to observe the following screenshot.
   ![Screenshot 2024-07-25 at 1 43 31 PM](https://github.com/user-attachments/assets/9b2c8dc2-7296-4bac-a119-d14f691f1050)

**Expected behavior**
Show the errors in some manner that does not require the user to scroll. Auto-scrolling would be OK, but might not be the best option.

**Software**

```
plone6-demo 1.0.0
Volto 17.16.0
Plone 6.0.11
plone.restapi 9.6.1
CMF 3.5
Zope 5.9
Python 3.11.9 (main, May 14 2024, 08:32:26) [GCC 10.2.1 20210110]
PIL 9.5.0 (Pillow)
```

Add-ons

```
@plone/volto-slate 17.16.0
volto-subblocks 2.0.0
volto-form-block 3.8.3
@eeacms/volto-accordion-block 10.4.6
@kitconcept/volto-button-block 2.3.1
@kitconcept/volto-heading-block 2.4.0
@kitconcept/volto-highlight-block 3.0.1
@kitconcept/volto-introduction-block 1.0.0
@kitconcept/volto-separator-block 4.1.1
@kitconcept/volto-slider-block 6.3.1
@kitconcept/volto-light-theme 3.3.1
volto-demo 0.1.0
```

**Comment by charanbhatia** [comment]
hi, can you please assign this issue to me. I want to work on this.

**Comment by stevepiercy** [comment]
@charanbhatia, no. Please read and follow [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do), [Contributing to Plone](https://6.docs.plone.org/contributing/index.html), and [Contributing to Volto](https://6.docs.plone.org/contributing/volto.html).

**Comment by barelysomethin** [comment]
hi @stevepiercy , i went through the volto repo but can't exactly the find the file which should contain this form field and code for control panel in general. can i get some guidance or any advice in general to navigate through the codebase efficiently.

**Comment by stevepiercy** [comment]
@barelysomethin I don't know.

Also please read and follow [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do), [Contributing to Plone](https://6.docs.plone.org/contributing/index.html), and [Contributing to Volto](https://6.docs.plone.org/contributing/volto.html).

**Comment by barelysomethin** [comment]
alright, thanks.

**Comment by wesleybl** [comment]
@barelysomethin the file in question is this one:

https://github.com/plone/volto/blob/main/packages/volto/src/components/manage/Controlpanels/Users/UsersControlpanel.jsx

**Comment by Srihari2222** [comment]
@stevepiercy  yeah, it is not obvious. I have a idea of adding tool tip so that the error can display right after submitting form without scrolling to top. Can you assign to me?

**Comment by stevepiercy** [comment]
@Srihari2222 no, we don't assign issues, but you can claim it for yourself. See [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially Item 1 in [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do).

**Comment by Shyam-Raghuwanshi** [comment]
hi @stevepiercy i think the toast would be a good approach?

**Comment by stevepiercy** [comment]
@Shyam-Raghuwanshi I'm not entirely sure, as I think toasts fade away after a few seconds. We need direction from people interested in UX on the @plone/volto-team.

I think I would prefer something that appears adjacent to the password field that updates as the user types, not only when submitted, such as https://6.docs.plone.org/storybook/?path=/story/edit-widgets-id--errored. I would like to see https://6.docs.plone.org/storybook/?path=/story/edit-widgets-password--password enhanced.

Let's wait for clear direction.

**Comment by Shyam-Raghuwanshi** [comment]
ok

**Comment by stevepiercy** [comment]
This is still an issue.

New information, when I hover over the Save button, the second black circle disappears.

### No hover over Save button
<img width="919" height="179" alt="Image" src="https://github.com/user-attachments/assets/a041c710-d332-42b8-9342-6fb4ed270957" />

### Hover over Save button
<img width="919" height="179" alt="Image" src="https://github.com/user-attachments/assets/7bbc6774-aeac-44db-af1d-b66f948fd6e1" />

**Comment by pratyush07-hub** [comment]
> [@Shyam-Raghuwanshi](https://github.com/Shyam-Raghuwanshi) I'm not entirely sure, as I think toasts fade away after a few seconds. We need direction from people interested in UX on the [@plone/volto-team](https://github.com/orgs/plone/teams/volto-team).
> 
> I think I would prefer something that appears adjacent to the password field that updates as the user types, not only when submitted, such as https://6.docs.plone.org/storybook/?path=/story/edit-widgets-id--errored. I would like to see https://6.docs.plone.org/storybook/?path=/story/edit-widgets-password--password enhanced.
> 
> Let's wait for clear direction.

Hello @stevepiercy  @wesleybl , I have two suggestions for improving the validation error handling here. 
1. we can show the validation error next to the specific field that is missing or invalid (for example, password, username, or email), so users can immediately see what needs to be fixed. 
2. In addition to that, we can also display a general error message above the Submit/Save button at the bottom of the form to clearly indicate that the submission failed, especially when multiple fields have errors.

**Comment by shivaansh0610-LUFFY** [comment]
Hi,

I’d like to work on improving the error display UX in this form.
From the reproduction steps, it looks like validation errors aren’t immediately visible (no scroll), and the submit button appears duplicated after a failed submission.

Before I start, is there any preferred or existing UX pattern in Volto for handling this kind of validation feedback (for example, auto-scrolling to errors, inline display, or both)?

**Comment by wesleybl** [comment]
@shivaansh0610-LUFFY @pratyush07-hub I think the most important thing here is to scroll the page so the user sees the error. I don't know if the backend response indicates which field is causing the error.
